crates/server-primitives/src/admin.rs

```diff
@@ -102,16 +112,34 @@ pub struct NearSignatureMessageMetadata {
 #[serde(rename_all = "camelCase")]
 pub struct EthSignatureMessageMetadata {}
 
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct StarknetSignatureMessageMetadata {}
+
 // Intermediate structs for initial parsing
 #[derive(Debug, Deserialize)]
 #[serde(rename_all = "camelCase")]
 pub struct IntermediateAddPublicKeyRequest {
-    pub wallet_signature: String,
+    pub wallet_signature: WalletSignature,
     pub payload: IntermediatePayload,
     pub wallet_metadata: WalletMetadata, // Reuse WalletMetadata as it fits the intermediate step
     pub context_id: Option<calimero_primitives::context::ContextId>,
 }
 
+#[derive(Debug, Deserialize)]
+#[serde(untagged)]
+pub enum WalletSignature {
+    String(String),
+    StarknetPayload(StarknetPayload),
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct StarknetPayload {
+    pub signature: Vec<String>,
```

Comment:
this is just 2 parameters, yeah?

maybe `[String, 2]` or split it into it's constituents `r` and `s`

Reply:
depending on the wallet, the Argent wallet returns 5 elements in the Vec while Metamask Snap returns 3, it isn't constant and is subjugated to a possible change in the future, so I implemented it like this.